core/extension/gdextension_interface.h

```diff
@@ -1526,6 +1526,21 @@ typedef void (*GDExtensionInterfaceStringOperatorPlusEqWcstr)(GDExtensionStringP
  */
 typedef void (*GDExtensionInterfaceStringOperatorPlusEqC32str)(GDExtensionStringPtr p_self, const char32_t *p_b);
 
+/* INTERFACE: StringName Utilities */
+
+/**
+ * @name string_name_new_with_utf8_chars
+ * @since 4.1
```

Comment:
This is probably actually Godot 4.2 material -- 4.1 is about to go into RC next week!